commands/container-registry/token/tokens.go

```diff
@@ -0,0 +1,184 @@
+	Status              string `json:"Status,omitempty"`
+}
+
+func getTokensRows(tokens *[]ionoscloud.TokenResponse) []map[string]interface{} {
+	out := make([]map[string]interface{}, 0, len(*tokens))
+	for _, token := range *tokens {
+		var tokenPrint TokenPrint
+		if idOk, ok := token.GetIdOk(); ok && idOk != nil {
+			tokenPrint.TokenId = *idOk
+		}
+		if propertiesOk, ok := token.GetPropertiesOk(); ok && propertiesOk != nil {
+			if displayNameOk, ok := propertiesOk.GetNameOk(); ok && displayNameOk != nil {
+				tokenPrint.DisplayName = *displayNameOk
+			}
+			if expiryDateOk, ok := propertiesOk.GetExpiryDateOk(); ok && expiryDateOk != nil {
+				tokenPrint.ExpiryDate = expiryDateOk.String()
+			}
+			if credentialsOk, ok := propertiesOk.GetCredentialsOk(); ok && credentialsOk != nil {
+				if usernameOk, ok := credentialsOk.GetUsernameOk(); ok && usernameOk != nil {
+					tokenPrint.CredentialsUsername = *usernameOk
+				}
+				if passwordOk, ok := credentialsOk.GetPasswordOk(); ok && passwordOk != nil {
+					tokenPrint.CredentialsPassword = *passwordOk
+				}
+			}
+			if statusOk, ok := propertiesOk.GetStatusOk(); ok && statusOk != nil {
+				tokenPrint.Status = *statusOk
+			}
+		}
+		o := structs.Map(tokenPrint)
+		out = append(out, o)
+	}
+	return out
+}
+
+var allCols = structs.Names(TokenPrint{})
+
+// TODO: check how other resources are doing this
+func TokensIds() []string {
+	client, _ := config.GetClient()
+	svc := resources.NewRegistriesService(client, context.Background())
+	regs, _, _ := svc.List("")
+
+	svcToken := resources.NewTokenService(client, context.Background())
+	regsIDs := *regs.GetItems()
+
+	var allTokens []ionoscloud.TokenResponse
+
+	for _, regID := range regsIDs {
+		tokens, _, _ := svcToken.List(*regID.GetId())
+		allTokens = append(allTokens, *tokens.GetItems()...)
+	}
+
+	return functional.Map(
+		allTokens, func(reg ionoscloud.TokenResponse) string {
+			return *reg.GetId()
+		},
+	)
+}
+
+func ParseExpiryTime(expiryTime string) (time.Duration, error) {
+	years := 0
+	months := 0
+	days := 0
+	hours := 0
+
+	if !strings.ContainsAny(expiryTime, "0123456789") {
+		return 0, fmt.Errorf("invalid expiry time format")
+	}
+
+	number := ""
+
+	for i := 0; i < len(expiryTime); i++ {
+		if string(expiryTime[i]) != "y" && string(expiryTime[i]) != "m" && string(expiryTime[i]) != "d" && string(expiryTime[i]) != "h" {
+			number += string(expiryTime[i])
+		} else if expiryTime[i] == 'y' {
+			years, _ = strconv.Atoi(number)
+			number = ""
+		} else if expiryTime[i] == 'm' {
+			months, _ = strconv.Atoi(number)
+			number = ""
+		} else if expiryTime[i] == 'd' {
+			days, _ = strconv.Atoi(number)
+			number = ""
+		} else if expiryTime[i] == 'h' {
+			hours, _ = strconv.Atoi(number)
+			number = ""
+		}
+	}
+
+	return time.Duration(years)*time.Hour*24*365 + time.Duration(months)*time.Hour*24*30 + time.Duration(days)*time.Hour*24 + time.Duration(hours)*time.Hour, nil
```

Comment:
why not parse the string using time pkg ? into a duration

Reply:
Valid time units for time.ParseDuration are "ns", "us" (or "µs"), "ms", "s", "m" or "h".
I have to parse manually larger time units.